openquake/calculators/base.py

```diff
@@ -242,7 +242,16 @@ def run(self, pre_execute=True, concurrent_tasks=None, remove=True,
                     self.pre_execute()
                 self.result = self.execute()
                 if self.result is not None:
-                    self.post_execute(self.result)
+                    try:
+                        self.post_execute(self.result)
+                    except RuntimeError as exc:
+                        if (len(self.oqparam.sites) == 1 and
```

Comment:
This is NOT the way to determine the number of sites. Use self.N instead. If it does not work, debug why.

Reply:
Also, I don't like wrapping the entire `post_execute`. What it is the line causing the error? Wrap only that line please.

Reply:
Done